sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -777,3 +777,54 @@ def test_copy_X_with_auto_gram():
     linear_model.lars_path(X, y, Gram='auto', copy_X=True, method='lasso')
     # X did not change
     assert_allclose(X, X_before)
+
+
+@pytest.mark.parametrize("LARS, has_coef_path, args",
+                         ((Lars, True, {}),
+                          (LassoLars, True, {}),
+                          (LassoLarsIC, False, {}),
+                          (LarsCV, True, {}),
+                          # max_iter=5 is for avoiding ConvergenceWarning
+                          (LassoLarsCV, True, {"max_iter": 5})))
+@pytest.mark.parametrize("x_data_type, y_data_type, expected_type", (
+    (np.float32, np.float32, np.float32),
+    (np.float64, np.float64, np.float64)))
```

Comment:
Why not just have 1 "dtype" variable parametrized over (np.float32, np.float64) ?

Reply:
I have changed to use one parameter.